src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -46,6 +45,10 @@ class TaskKiller @Inject() (
             await(Future.sequence(foundInstances.map(i => instanceTracker.setGoal(i.instanceId, Goal.Decommissioned, GoalChangeReason.UserRequest)))): @silent
             await(expunge(foundInstances)): @silent
             await(instancesAreTerminal): @silent
+            // We've wiped instances *without scaling*, hence we have to launch replacements for them. Note that this
+            // is not the case if wipe=false, here only the tasks are killed but instances survive.
+            logger.info(s"Successfully wiped instances: ${foundInstances.map(_.instanceId).mkString(",")}. Now launching ${foundInstances.size} replacement instances.")
+            instanceTracker.schedule((0 until foundInstances.size).map(_ => Instance.scheduled(runSpec)))
```

Comment:
this is not correct for residents.

Reply:
also - what if there's a deployment ongoing? Shouldn't the deployment then handle this?

Reply:
I thought that when we're in the middle of deployment and decommission (wipe) that should be received by the orchestrator and the orchestrator should handle it ? as it would a task failure in between deployment....

Reply:
@alenkacz is right, we can't just schedule new instances here because of a possible on-going deployment. I put it in there mostly to see if anything major breaks - it didn't! Now, it's a message to the `MarathonSchedulerActor` which strikes me as somewhat hacky but that's c'est la vie